### ISSUE TITLE
org.elasticsearch.add_retention_lease.RetentionLeaseBwcIT.testRetentionLeaseBwcIT Test failure

### ISSUE THREAD

**Issue by benwtrent** [elastic/elasticsearch#40559]
testRetentionLeaseBwcIT has caused a handful of 6.7 BWC builds to fail.
Failure reproduces locally
Reproduce lines: 
```
./gradlew :qa:retention-lease-bwc:v6.7.0#newClusterTestRunner -Dtests.seed=7617A5AA63CC7356 -Dtests.class=org.elasticsearch.add_retention_lease.RetentionLeaseBwcIT -Dtests.method="testRetentionLeaseBwcIT" -Dtests.security.manager=true -Dtests.locale=pl -Dtests.timezone=America/Grand_Turk -Dtests.distribution=zip -Dcompiler.java=12 -Druntime.java=8
```

Failure stack trace:
```
4:13:11    > Throwable #1: java.net.SocketTimeoutException
14:13:11    > 	at __randomizedtesting.SeedInfo.seed([7617A5AA63CC7356:BFE88988248466E5]:0)
14:13:11    > 	at org.elasticsearch.client.RestClient$SyncResponseListener.get(RestClient.java:944)
14:13:11    > 	at org.elasticsearch.client.RestClient.performRequest(RestClient.java:233)
14:13:11    > 	at org.elasticsearch.test.rest.ESRestTestCase.ensureGreen(ESRestTestCase.java:786)
14:13:11    > 	at org.elasticsearch.add_retention_lease.RetentionLeaseBwcIT.testRetentionLeaseBwcIT(RetentionLeaseBwcIT.java:81)
14:13:11    > 	at java.lang.Thread.run(Thread.java:748)
14:13:11    > Caused by: java.net.SocketTimeoutException
14:13:11    > 	at org.apache.http.nio.protocol.HttpAsyncRequestExecutor.timeout(HttpAsyncRequestExecutor.java:375)
14:13:11    > 	at org.apache.http.impl.nio.client.InternalIODispatch.onTimeout(InternalIODispatch.java:92)
14:13:11    > 	at org.apache.http.impl.nio.client.InternalIODispatch.onTimeout(InternalIODispatch.java:39)
14:13:11    > 	at org.apache.http.impl.nio.reactor.AbstractIODispatch.timeout(AbstractIODispatch.java:175)
14:13:11    > 	at org.apache.http.impl.nio.reactor.BaseIOReactor.sessionTimedOut(BaseIOReactor.java:263)
14:13:11    > 	at org.apache.http.impl.nio.reactor.AbstractIOReactor.timeoutCheck(AbstractIOReactor.java:492)
14:13:11    > 	at org.apache.http.impl.nio.reactor.BaseIOReactor.validate(BaseIOReactor.java:213)
14:13:11    > 	at org.apache.http.impl.nio.reactor.AbstractIOReactor.execute(AbstractIOReactor.java:280)
14:13:11    > 	at org.apache.http.impl.nio.reactor.BaseIOReactor.execute(BaseIOReactor.java:104)
14:13:11    > 	at org.apache.http.impl.nio.reactor.AbstractMultiworkerIOReactor$Worker.run(AbstractMultiworkerIOReactor.java:588)
14:13:11    > 	... 1 more
14:13:11 Completed [1/1] in 34.59s, 1 test, 1 error <<< FAILURES!
```

Running locally, there is nothing in the node logs (new and old) and indicate a start up failure. The new node spins up just fine, and is added to the cluster. 

Investigating further.

**Comment by benwtrent** [comment]
Attempted again after a `./gradlew clean` locally and it failed at a different location

```
> Throwable #1: org.elasticsearch.client.ResponseException: method [PUT], host [http://127.0.0.1:63711], URI [/index/_add_retention_lease?retaining_sequence_number=-1&id=test-1], status line [HTTP/1.1 405 Method Not Allowed]
   > {"error":"Incorrect HTTP method for uri [/index/_add_retention_lease?retaining_sequence_number=-1&id=test-1] and method [PUT], allowed: [POST]","status":405}
   >    at __randomizedtesting.SeedInfo.seed([7617A5AA63CC7356:BFE88988248466E5]:0)
   >    at org.elasticsearch.client.RestClient$SyncResponseListener.get(RestClient.java:936)
   >    at org.elasticsearch.client.RestClient.performRequest(RestClient.java:233)
   >    at org.elasticsearch.add_retention_lease.RetentionLeaseBwcIT.testRetentionLeaseBwcIT(RetentionLeaseBwcIT.java:96)
   >    at java.lang.Thread.run(Thread.java:748)
```

So, maybe not as repeatable as I initially though :/

**Comment by jasontedor** [comment]
Closed by #40562
